### PR TITLE
Addresses 33 CA1801 issues on internal APIs on `src/NuGet.Core` folder - Part 1

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1330

Regression? Last working version:

## Description

- Reviews and fixes 31 CA1801 issues. More details on table below.
- Added tests on issues when the parameter is used in the method.

Description | Notes | Project | File | Line | Column | In PR? | Added Tests?
-- | -- | -- | -- | -- | -- | -- | --
Parameter   log of method GetInstalledPackageReferences is never used. Remove the   parameter or use it in the method body. | Removed param | NuGet.Build.Tasks (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Build.Tasks\BuildTasksUtility.cs | 621 | 170 | Yes | No
Parameter   restorePC of method RestoreAsync is never used. Remove the parameter or use   it in the method body. | It's already used | NuGet.Build.Tasks   (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Build.Tasks\BuildTasksUtility.cs | 163 | 18 | Yes | No
Parameter   project of method GetItemGroup is never used. Remove the parameter or use it   in the method body. | Removed param | NuGet.CommandLine.XPlat (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.CommandLine.XPlat\Utility\MSBuildAPIUtility.cs | 194 | 69 | Yes | No
Parameter   project of method GetItemGroup is never used. Remove the parameter or use it   in the method body. | Removed param | NuGet.CommandLine.XPlat   (netcoreapp2.1) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.CommandLine.XPlat\Utility\MSBuildAPIUtility.cs | 194 | 69 | Yes | No
Parameter   builder of method AddOutputFiles is never used. Remove the parameter or use   it in the method body. | Removed param | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\MSBuildProjectFactory.cs | 119 | 52 | Yes | No
Parameter   builder of method AddOutputFiles is never used. Remove the parameter or use   it in the method body. | Removed param | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\MSBuildProjectFactory.cs | 119 | 52 | Yes | No
Parameter   lockFileLib of method GetContentGroupsForFramework is never used. Remove the   parameter or use it in the method body. | Already removed | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\ContentFiles\ContentFileUtils.cs | 28 | 35 | No | No
Parameter   framework of method GetContentFileGroup is never used. Remove the parameter   or use it in the method body. | Removed param | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\ContentFiles\ContentFileUtils.cs | 74 | 28 | Yes | No
Parameter   lockFileLib of method GetContentGroupsForFramework is never used. Remove the   parameter or use it in the method body. | Already removed | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\ContentFiles\ContentFileUtils.cs | 28 | 35 | No | No
Parameter   framework of method GetContentFileGroup is never used. Remove the parameter   or use it in the method body. | Removed param | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\ContentFiles\ContentFileUtils.cs | 74 | 28 | Yes | No
Parameter   range of method GetSourceInfosForIdAsync is never used. Remove the parameter   or use it in the method body. | Removed param | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\Diagnostics\UnresolvedMessages.cs | 199 | 26 | Yes | No
Parameter   range of method GetSourceInfosForIdAsync is never used. Remove the parameter   or use it in the method body. | Removed param | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\Diagnostics\UnresolvedMessages.cs | 199 | 26 | Yes | No
Parameter   parentId of method .ctor is never used. Remove the parameter or use it in the   method body. | Fixed | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\OriginalCaseGlobalPackageFolder.cs | 37 | 77 | Yes | Yes
Parameter   parentId of method .ctor is never used. Remove the parameter or use it in the   method body. | Fixed | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\OriginalCaseGlobalPackageFolder.cs | 37 | 77 | Yes | Yes
Parameter   token of method WalkDependenciesAsync is never used. Remove the parameter or   use it in the method body. | Used token | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\ProjectRestoreCommand.cs | 254 | 31 | Yes | Yes
Parameter   token of method WalkDependenciesAsync is never used. Remove the parameter or   use it in the method body. | Used token | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\ProjectRestoreCommand.cs | 254 | 31 | Yes | Yes
Parameter   context of method GetProjectReferences is never used. Remove the parameter or   use it in the method body. | Removed param | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\RestoreCommand.cs | 1064 | 87 | Yes | No
Parameter   lockFile of method DowngradeLockFileToV2 is never used. Remove the parameter   or use it in the method body. | Already removed method | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\RestoreCommand.cs | 1160 | 53 | No | No
Parameter   context of method GetProjectReferences is never used. Remove the parameter or   use it in the method body. | Removed param | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\RestoreCommand.cs | 1064 | 87 | Yes | No
Parameter   lockFile of method DowngradeLockFileToV2 is never used. Remove the parameter   or use it in the method body. | Removed method | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\RestoreCommand.cs | 1160 | 53 | No | No
Parameter   token of method CommitAssetsFileAsync is never used. Remove the parameter or   use it in the method body. | Used parameter | NuGet.Commands (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\RestoreResult.cs | 184 | 31 | Yes | Yes
Parameter   token of method CommitAssetsFileAsync is never used. Remove the parameter or   use it in the method body. | Used parameter | NuGet.Commands (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Commands\RestoreCommand\RestoreResult.cs | 184 | 31 | Yes | Yes
Parameter   mappings of method GetTable is never used. Remove the parameter or use it in   the method body. | Removed param | NuGet.Frameworks (net40) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Frameworks\CompatibilityTable.cs | 74 | 148 | Yes | No
Parameter   mappings of method GetTable is never used. Remove the parameter or use it in   the method body. | Removed param | NuGet.Frameworks (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Frameworks\CompatibilityTable.cs | 74 | 148 | Yes | No
Parameter   mappings of method GetTable is never used. Remove the parameter or use it in   the method body. | Removed param | NuGet.Frameworks   (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Frameworks\CompatibilityTable.cs | 74 | 148 | Yes | No
Parameter   includeOptional of method ExplodePortableFrameworks is never used. Remove the   parameter or use it in the method body. | Removed param | NuGet.Frameworks (net40) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Frameworks\FrameworkReducer.cs | 382 | 138 | Yes | No
Parameter   includeOptional of method ExplodePortableFrameworks is never used. Remove the   parameter or use it in the method body. | Removed param | NuGet.Frameworks (net472) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Frameworks\FrameworkReducer.cs | 382 | 138 | Yes | No
Parameter   includeOptional of method ExplodePortableFrameworks is never used. Remove the   parameter or use it in the method body. | Removed param | NuGet.Frameworks   (netstandard2.0) | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.Frameworks\FrameworkReducer.cs | 382 | 138 | Yes | No
Parameter   log of method PreviewRestoreAsync is never used. Remove the parameter or use   it in the method body. | d81903a8 introduces new logging param. Never used since then. Removed | NuGet.PackageManagement | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.PackageManagement\BuildIntegration\DependencyGraphRestoreUtility.cs | 150 | 21 | Yes | No
Parameter   token of method PreviewRestoreAsync is never used. Remove the parameter or   use it in the method body. | Already used | NuGet.PackageManagement | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.PackageManagement\BuildIntegration\DependencyGraphRestoreUtility.cs | 151 | 31 | Yes | No
Parameter   cancellationToken of method TryAddFileAsync is never used. Remove the   parameter or use it in the method body. | Fixed | NuGet.PackageManagement | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.PackageManagement\Utility\MSBuildNuGetProjectSystemUtility.cs | 85 | 31 | Yes | Yes
Parameter   cancellationToken of method DeleteFileSafeAsync is never used. Remove the   parameter or use it in the method body. | Fixed | NuGet.PackageManagement | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.PackageManagement\Utility\MSBuildNuGetProjectSystemUtility.cs | 364 | 31 | Yes | Yes
Parameter   token of method GetNupkgPath is never used. Remove the parameter or use it in   the method body. | Fixed | NuGet.PackageManagement | C:\repos\NuGet.Client\src\NuGet.Core\NuGet.PackageManagement\Utility\PackagesConfigContentHashProvider.cs | 37 | 88 | Yes | Yes

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added: Added unit tests
  - ~[ ] Test exception~
  - ~[ ] N/A~

- **Documentation**
  - ~[ ] Documentation PR or issue filled~
  - [x] N/A: No major changes to NuGet

### CODE REVIEW

File: src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs [hunk 1]
```diff
@@ -147,13 +147,12 @@ public static string GetDefaultDGSpecFileName()
...
             IEnumerable<SourceRepository> sources,
             Guid parentId,
-            ILogger log,
             CancellationToken token)
         {
             token.ThrowIfCancellationRequested();
 
             // Restoring packages
-            var logger = context.Logger;
+            _ = context.Logger;
```
Comment (jeffkl) [inline src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs]: You should just remove this line right?

**Review by jeffkl — APPROVED** [review]

File: test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
   <ItemGroup>
     <ProjectReference Include="$(TestUtilitiesDirectory)Test.Utility\Test.Utility.csproj" />
+    <ProjectReference Include="..\..\..\src\NuGet.Clients\NuGet.VisualStudio.Common\NuGet.VisualStudio.Common.csproj" />
```
Comment (nkolev92) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: I don't think we should do this. 

NuGet.Commands is a core level test project and shouldn't depend on the VS layer.

**Review by nkolev92 — COMMENTED** [review]
Only one objection. 

The rest looks great!

File: test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj (hunk 2 above)
Comment (dominoFire) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: I believe I have a telemetry dependency on this. Taking a look.

Comment (dominoFire) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: I think I can move the test to NuGet.VisualStudio.Common.Test and avoid this project dependency.

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: Sounds good. 

Re-request review when ready.

Comment (dominoFire) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: Ended up keeping the test in the same class. I mocked telemetry interface to get rid of telemetry service provided by N.VS.Common

File: src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs (hunk 1 above)
Comment (dominoFire) [reply src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs]: Yep. Removed.

File: test/NuGet.Core.Tests/NuGet.PackageManagement.Test/Utility/MSBuildNuGetProjectSystemUtilityTests.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            });
+        }
+
+        [Fact]
+        public async Task DeleteFileSafeAsync_WithCancellationToken_ThrowsAsync()
+        {
+            await Assert.ThrowsAsync<OperationCanceledException>(async () =>
+            {
+                await MSBuildNuGetProjectSystemUtility.DeleteFileSafeAsync(
+                    path: "",
```
Comment (kartheekp-ms) [inline test/NuGet.Core.Tests/NuGet.PackageManagement.Test/Utility/MSBuildNuGetProjectSystemUtilityTests.cs]: nit
``` suggestion
                    path: string.Empty,
```

File: test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommand/OriginalCaseGlobalPackageFolderTests.cs [hunk 4]
```diff
@@ -178,6 +180,52 @@ public async Task CopyPackagesToOriginalCaseAsync_OnlyInstallsPackagesOnceAsync(
...
+                var target = new OriginalCaseGlobalPackageFolder(request, parentIdGuid);
+
+                // Act
+                await target.CopyPackagesToOriginalCaseAsync(
+                    new[] { graphA, graphB },
+                    CancellationToken.None);
+            }
+
+            // Assert
+            Assert.Equal(1, telemetryEvents.Where(x => x.Name == "PackageExtractionInformation").Count());
```
Comment (kartheekp-ms) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommand/OriginalCaseGlobalPackageFolderTests.cs]: nit
``` suggestion
            Assert.True(telemetryEvents.Where(x => x.Name == "PackageExtractionInformation").Any());
```

**Review by kartheekp-ms — APPROVED** [review]

File: test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommand/OriginalCaseGlobalPackageFolderTests.cs (hunk 4 above)
Comment (dominoFire) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommand/OriginalCaseGlobalPackageFolderTests.cs]: I want to assert for exactly one event.

**Review by nkolev92 — APPROVED** [review]

File: test/NuGet.Core.Tests/NuGet.PackageManagement.Test/Utility/MSBuildNuGetProjectSystemUtilityTests.cs (hunk 3 above)
Comment (dominoFire) [reply test/NuGet.Core.Tests/NuGet.PackageManagement.Test/Utility/MSBuildNuGetProjectSystemUtilityTests.cs]: Fixed.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommand/OriginalCaseGlobalPackageFolderTests.cs (hunk 4 above)
Comment (dominoFire) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommand/OriginalCaseGlobalPackageFolderTests.cs]: Added  `Assert.Single()`

**Pull request merged**